### PR TITLE
RFC: npm pkg

### PR DESCRIPTION
**Pull request by ruyadorno**
[See rendered RFC](https://github.com/npm/rfcs/blob/dd01b3d3aa54682c32ef1a0d329afa1ef528eaf7/accepted/0000-npm-pkg.md)

### CODE REVIEW

File: accepted/0000-npm-pkg.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+$ npm pkg get scripts
+{
+  "lint": "eslint"
+}
+$ npm pkg get scripts.lint
+"eslint"
+$ npm pkg set scripts.lint.eslint.foo=bar
+error: "scripts.lint" exists and is not an object
+$ npm pkg delete scripts
+$ npm pkg set scripts.lint.eslint.foo=bar
```
Comment (ljharb) [inline accepted/0000-npm-pkg.md:44]: any reason this couldn't be combined into, like, `npm pkg set -f scripts.lint.eslint.foo=bar`?

File: accepted/0000-npm-pkg.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    }
+  }
+}
+$ npm pkg set tap.timeout=60 --pkg-type=number
+$ npm pkg get tap
+{
+  "timeout": 60
+}
+$ npm pkg set contributors[0].name=Ruy contributors[0].email=foo@bar.ca
+$ npm pkg set contributors[1].name=Gar contributors[1].email=foo@bar.ca
```
Comment (ljharb) [inline accepted/0000-npm-pkg.md:59]: an example of how to add multiple items to the array in one command would be nice :-) (like, an "array push" example)

Comment (ruyadorno) [reply accepted/0000-npm-pkg.md:59]: I really dig your `[]` idea but that was not part of the `npm view` syntax, so I haven't brought it into it yet 😁 but I'm still considering adding it to the parser!

now adding multiple items at once might be tricky... I really think it's a good idea to steer away from handling objects (like we discussed in the last meeting) so at first the idea is to only handle one literal value at a time

File: accepted/0000-npm-pkg.md (hunk 1 above)
Comment (ruyadorno) [reply accepted/0000-npm-pkg.md:44]: I like that idea! 😄

File: accepted/0000-npm-pkg.md (hunk 2 above)
Comment (ljharb) [reply accepted/0000-npm-pkg.md:59]: `npm pkg set contributors[].name=Bob contributors[].name=Jan` to add two new objects with the respective names?

Comment (ruyadorno) [reply accepted/0000-npm-pkg.md:59]: as long as that new `[]` syntax lands, yeah it should!

Comment (ruyadorno) [reply accepted/0000-npm-pkg.md]: @ljharb just to make sure you're aware, this syntax has not yet landed in this initial implementation but I'm looking forward to seeing it land in the future 😄

Comment (ruyadorno) [reply accepted/0000-npm-pkg.md]: actually added it for `npm pkg set` here: https://github.com/npm/cli/pull/3536

**Pull request closed without merging**